src/mpi/comm/commutil.c

```diff
@@ -602,6 +602,13 @@ int MPIR_Comm_create_subcomms(MPIR_Comm * comm)
         comm->node_comm->local_size = num_local;
         comm->node_comm->remote_size = num_local;
 
+        /* Copy relevant hints to node_comm */
+        comm->node_comm->hints[MPIR_COMM_HINT_SENDER_VCI] = comm->hints[MPIR_COMM_HINT_SENDER_VCI];
+        comm->node_comm->hints[MPIR_COMM_HINT_RECEIVER_VCI] =
+            comm->hints[MPIR_COMM_HINT_RECEIVER_VCI];
+        comm->node_comm->hints[MPIR_COMM_HINT_VCI] = comm->hints[MPIR_COMM_HINT_VCI];
+
+
```

Comment:
Wrap them in a static function e.g. `subcomm_inherit_hints(MPIR_Comm *parent_comm)`